packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/JSONResourceFactory.java

```diff
@@ -21,18 +21,47 @@
 import org.eclipse.sirius.emfjson.resource.JsonResourceImpl;
 
 /**
- * Special factory for {@link JsonResource} for Sirius Web.</br>
+ * Special factory for {@link JsonResource}.</br>
  * This factory ensures that objects in that resource are identified by their uuid.
  *
  * @author lfasani
  */
-public class SiriusWebJSONResourceFactoryImpl extends ResourceFactoryImpl {
+public class JSONResourceFactory extends ResourceFactoryImpl {
 
     @Override
     public JsonResource createResource(URI uri) {
+        if (uri == null || uri.scheme() == null || uri.scheme().isBlank()) {
+            throw new IllegalArgumentException(String.format("Missing scheme for URI %s", uri)); //$NON-NLS-1$
```

Comment:
The explicit `throw` makes ArchUnit tests fail; should be rewritten to hide the throw (e.g. using an `Objects.requireNonNull` or `Optional#orElseThrow`...).